server/src/main/java/org/opensearch/action/search/AbstractSearchAsyncAction.java

```diff
@@ -637,6 +641,12 @@ public SearchPhase getCurrentPhase() {
 
     private void setCurrentPhase(SearchPhase phase) {
         currentPhase = phase;
+        // The WrappingSearchAsyncActionPhase (see please CanMatchPreFilterSearchPhase as one example) is a special case
+        // of search phase that wraps SearchAsyncActionPhase as SearchPhase. The AbstractSearchAsyncAction manages own
+        // onPhaseStart / onPhaseFailure / OnPhaseDone callbacks and the wrapping SearchPhase is being abandoned
+        // (fe, has no onPhaseEnd callbacks called ever). To fix that, we would not send any notifications for this
+        // phase.
+        currentPhaseHasLifecycle = !(phase instanceof WrappingSearchAsyncActionPhase);
```

Comment:
Can this be `(phase instanceof WrappingSearchAsyncActionPhase == false)`?

Reply:
Sure!